Add docs for service accounts 8.5 (#46801)
grafana/grafana: docs/sources/administration/service-accounts/about-service-accounts.md

---
title: About service accounts
menuTitle: About service accounts
description: 'This page contains detailed information about service accounts in Grafana'
weight: 30
---

# About service accounts in Grafana

A service account can be used to run automated or compute workloads. Applications use service account tokens to authorize themselves as a service account.

> **Note:** Service accounts are available in Grafana 8.5+ as a beta feature, to enable service accounts refer to [Enable service accounts]({{< relref "./enable-service-accounts.md#">}}) section.

A common use case for creating a service account is to perform operations on automated or triggered tasks. You can use service accounts to:

- Schedule reports for specific dashboards to be delivered on a daily/weekly/monthly basis
- Define alerts in your system to be used in Grafana
- Set up an external authentication provider to manage users and permissions across an organization
- Establish machine-to-machine communication
- Interact with Grafana without logging in as a user

> You can also use service accounts in combination with fine-grained access control to grant users and teams specific scopes.

Reviewer: Since we don't have yet mapping of a service account to teams, I'd suggest to not mention teams here.

You can associate a service account with multiple tokens. For example:

- each team member can generate their own token

> -

Reviewer: Not sure if this is something unintentional here? The formatting is broken and kind of starts and ends in a strange way.

Author: unintentional. also added some more examples

> We recommend starting off by creating one service account per use case.

Reviewer 2: ```suggestion
We recommend the you begin by creating one service account for each use case.
```

> **Note:** Service accounts can only act in the organization they are created for. If you have the same task that is needed for multiple organizations, we recommend creating service accounts in each organization.

---

### Service accounts benefits

> When creating a token for a service account, we call them service account token; which are essentially a API key. The main difference to API keys is that permission is set on top of the API key when creating API keys, where as a service account token gets their permission from the service account.

Reviewer: I think it is worth to have a separate header explaining what are service account tokens, as in the benefits section we are essentially introducing a new concept in a confusing way. 

For example,

```
## Service account tokens

When you create a service account, you can associate one or more access tokens with it. You can use service access tokens the same way as API Keys, for example to access Grafana HTTP API programatically. 

Service account access tokens inherit permissions from service account directly.
```

Reviewer 2: +1 to Vardan's suggestion.

The added benefits of service accounts to API keys are:

> - Service accounts act like users in Grafana and can be enabled/disabled, granted specific permissions, and remain active until deleted or disabled whereas API keys only live until their expiry date.

Reviewer 2: ```suggestion
- Service accounts resemble Grafana users and can be enabled/disabled, granted specific permissions, and remain active until they are deleted or disabled. API keys are only valid until their expiry date.
```

- Service accounts can be associated with multiple tokens.
- Unlike API keys, service account tokens are not associated with a specific user, which means that applications can be authenticated even if a Grafana user is deleted.

> - You can grant granular permissions to service accounts by leveraging FGAC, API keys are not a part of FGAC. For more information about permissions, refer to [About users and permissions]({{< relref "../manage-users-and-permissions/about-users-and-permissions.md#">}}).

Reviewer: nit: expand FGAC to fine-grained access control and add a link

> - You can grant granular permissions to service accounts by leveraging FGAC, API keys are not a part of FGAC. For more information about permissions, refer to [About users and permissions]({{< relref "../manage-users-and-permissions/about-users-and-permissions.md#">}}).

Reviewer 2: ```suggestion
- You can grant granular permissions to service accounts by leveraging FGAC. API keys do not make use of FGAC. For more information about permissions, refer to [About users and permissions]({{< relref "../manage-users-and-permissions/about-users-and-permissions.md#">}}).
```

Reviewer 2: The link you are sending readers to is a general topic about OSS permissions. Do you intend to send them to the FGAC content?

Author: disregarded these infavor of making some of my own >D

---
title: About service accounts
menuTitle: About service accounts
aliases: [docs/sources/manage-service-accounts/_index.md, 
docs/sources/administration/service-accounts/about-service-accounts.md,
docs/sources/administration/service-accounts/create-service-accounts.md]
description: "This contains detailed information about service accounts in Grafana"
weight: 30
---

> # About service accounts in Grafana

Author: 👍

> A service account is a regular user account used to run automated or compute workloads which are intended for programmatic use either internally for a Grafana feature or to communicate outside of the Grafana environment. Applications use service account tokens to authorize themselves as a service account.

Reviewer 2: Update this sentence after you are settled on the change to the _index file.

> A service account is a regular user account used to run automated or compute workloads which are intended for programmatic use either internally for a Grafana feature or to communicate outside of the Grafana environment. Applications use service account tokens to authorize themselves as a service account.

Author: yea, valid comment. just enough information and technical detail that it is a user. 👋

A common use case for creating a service account is to perform operations on automated or triggered tasks. You can use service accounts to:

> - Schedule reports for specific dashboards to be delivered on a daily/weekly/monthly basis

Author: https://raintank-corp.slack.com/archives/C01JEKKTMCK/p1649242550307399

- Define alerts in your system to be used in Grafana

> - Set up to manage users and permissions across the organization to an external auth provider

Reviewer 2: ```suggestion
- Set up an external authentication provider to manage users and permissions across an organization
```

- Machine to machine communication

> It can also be used in combination with FGAC to grant specific scopes.

Reviewer 2: ```suggestion
You can also use service accounts in combination with fine-grained access control to grant users and teams specific scopes.
```

Reviewer 2: And then add a cross reference to the FGAC docs. "For more information about fine-grained access control, refer to xxx."

> 

Reviewer 2: This first sentence is passive voice and we want to write in the active voice. In this case, active voice is "You can associate a service account with multiple API keys." When would a user associate a service account with multiple API keys? Because you've introduced that this can be done, you should provide some examples/use cases of when to do it, or at least some benefits.

Author: gotcha. more direct and introduce examples perhaps

> A service account can be associated with multiple api keys. As a result, we recommend starting off by creating one service account per use case.

Author: @chri2547 . Wdyt?

My opinon is that we use "tokens" when it comes to service accounts refering to the "api keys" being added to the service accounts. As that is what we want to have going forward.

> A service account can be associated with multiple api keys. As a result, we recommend starting off by creating one service account per use case.

Author: gonna use the term "token" for service accounts instead

> A service account can be associated with multiple api keys. As a result, we recommend starting off by creating one service account per use case.

Reviewer 2: I agree, we should use the term 'token' for service accounts consistently.

> > **Note:** Service accounts act on an organization level, if you have the same task that is needed for multiple organizations, we recommend provisioning service accounts for each organization.

Reviewer 2: what do you mean by 'act on an organization level'?

---

> ## Grafana Service account vs API key

Reviewer 2: ```suggestion
## Grafana service account vs API key
```

> We have created service accounts for improved access control of programmatic access to Grafana. A service account token is a token that is used to authenticate a service account to Grafana and are very similar to API keys.

Reviewer 3: ```suggestion
We have created service accounts for improved access control of programmatic access to Grafana. A service account token is a token that is used to authenticate a service account to Grafana and is very similar to an API key.
```

> We recommend to start using service accounts over API keys.

Reviewer 2: ```suggestion
We recommend that you use service accounts instead of using API keys.
```

> ### Service accounts benefits over API keys

Reviewer 2: I would actually stop talking about API keys moving forward, and just go right to the benefits of service accounts.

> ### Service accounts benefits over API keys

Reviewer 2: ```suggestion
### Service accounts benefits
```

> When creating a token for a service account, we call them service account token; which are essentially a API key. The main difference to API keys is that permission is set on top of the API key when creating API keys, where as a service account token gets their permission from the service account.

Author: awesome! removed the whole section of comparison.

The added benefits of service accounts to API keys are:

> - Service accounts act like users in Grafana and can be disabled/enabled and granted permissions.

Reviewer 2: ```suggestion
- Service accounts act like users in Grafana and can be enabled, disabled, and granted permissions. You cannot grant permissions to API keys.
```

> - Service accounts can be associated with multiple api keys.

Reviewer 2: For each one of the service account benefits, tell me about the shortcomings of using just API keys. So for this one, some wording could be something like:

- You can associate a service accounts with multiple API keys, which provides greater flexibility when you want to xxxx (add something here about the limitation of using just API keys)

Author: 👍

> - Service account tokens are not tied to a specific user, therefore making sure applications can be authenticated even if a Grafana user is deleted.

Reviewer 2: ```suggestion
- Unlike API keys, service account tokens are not associated with a specific user, which means that applications can be authenticated even if a Grafana user is deleted.
```

> - Service accounts can be granted granular permissions. More information on specific permission refer to [About users and permissions]({{< relref "../manage-users-and-permissions/about-users-and-permissions.md#">}}).

Reviewer 2: ```suggestion
- You can grant granular permissions to service accounts. For more information about permissions, refer to [About users and permissions]({{< relref "../manage-users-and-permissions/about-users-and-permissions.md#">}}).
```

> - Service accounts can be granted granular permissions. More information on specific permission refer to [About users and permissions]({{< relref "../manage-users-and-permissions/about-users-and-permissions.md#">}}).

Reviewer 2: Here I'd like to know how API keys are lacking. Can a user not associate API keys with granular permissions? If not, what does that limitation mean in practical terms to the user? Also, you are linking the user to the standard permissions docs. Did you intend to send them to the FGAC docs which deal with more granular permissions? Just a question, perhaps you intend to send to the About permissions topic, which is fine.

> - Eventually API keys will be deprecated

Reviewer 2: I'm unsure if we should say that API keys will be deprecated, because it is so far in the future. This also creates tech debt because you'll need to remember to go back and remove this point after deprecation. I would remove this bullet point entirely.

Author: removed
